### PR TITLE
🎨 Fix afterOptimizeChunkAssets deprecation warning

### PR DESCRIPTION
**Pull request by legitatx**
- Remove legacy code and ran Prettier to clean the codebase. Fixes deprecation warning relating to afterOptimizeChunkAssets.

### CODE REVIEW

File: src/ExtensionReloader.ts [hunk 1]
```diff
@@ -116,11 +110,7 @@ export default class ExtensionReloaderImpl extends AbstractPluginReloader
   }
 
   public apply(compiler: Compiler) {
-    if (
-      (this._isWebpackGToEV4()
-        ? compiler.options.mode
-        : process.env.NODE_ENV) === "development"
-    ) {
+    if (compiler.options.mode === "development") {
```
Comment (rushilsrivastava) [inline src/ExtensionReloader.ts:113]: We should probably retain the `process.env.NODE_ENV` check here.

File: src/webpack/CompilerEventsFacade.ts [hunk 2]
```diff
@@ -5,36 +5,36 @@ export default class CompilerEventsFacade {
...
-        )
-      : this._compiler.hooks.compilation.tap(
-          CompilerEventsFacade.extensionName,comp =>
-            comp.hooks.afterOptimizeChunkAssets.tap(
-              CompilerEventsFacade.extensionName,chunks => call(comp, chunks),
-            ),
+    return this._compiler.hooks.compilation.tap(
+      CompilerEventsFacade.extensionName,
+      (comp) => {
+        const afterOptimizeChunkAssets = (chunks) => {
```
Comment (rushilsrivastava) [inline src/webpack/CompilerEventsFacade.ts:18]: Can we rename this function so there is no namespace overlap? 

Additionally, I am not sure if this passes in the same chunks via the tap, is there a reason we are taking this approach?

File: src/webpack/CompilerEventsFacade.ts [hunk 3]
```diff
@@ -5,36 +5,36 @@ export default class CompilerEventsFacade {
...
+          call(comp, chunks);
+        };
+        /* https://github.com/webpack/webpack/blob/main/lib/Compilation.js#L772-L779
+        afterOptimizeChunkAssets = PROCESS_ASSETS_STAGE_OPTIMIZE + 1
+        afterOptimizeChunkAssets = 101
+        */
+        const stage = 101;
```
Comment (rushilsrivastava) [inline src/webpack/CompilerEventsFacade.ts:25]: How are we getting 101 here? Additionally, we shouldn't hard code this but leave it referenceable.

You can just get it from `Compilation`.

```const stage = Compilation.PROCESS_ASSETS_STAGE_OPTIMIZE + 1```

**Review by rushilsrivastava — CHANGES_REQUESTED** [review]
Just some minor changes requested, looks good from here!

**Comment by legitatx** [comment]
> Just some minor changes requested, looks good from here!

Sweet, will get on this now!

File: src/webpack/CompilerEventsFacade.ts (hunk 3 above)
Comment (legitatx) [reply src/webpack/CompilerEventsFacade.ts:25]: I had just hardcoded the value here for debugging and forgot to reverse the change to make sure I was going about this the right way. Fixed now!

File: src/webpack/CompilerEventsFacade.ts [hunk 4]
```diff
@@ -5,36 +5,35 @@ export default class CompilerEventsFacade {
...
+        const afterOptimizeChunkAssets = (chunks) => {
+          call(comp, chunks);
+        };
+        /* https://github.com/webpack/webpack/blob/main/lib/Compilation.js#L772-L779
+        afterOptimizeChunkAssets = PROCESS_ASSETS_STAGE_OPTIMIZE + 1
+        */
+        const stage = Compilation.PROCESS_ASSETS_STAGE_OPTIMIZE + 1;
+        (comp.hooks as any).processAssets.tap(
+          { name: CompilerEventsFacade.extensionName, stage },
+          afterOptimizeChunkAssets,
```
Comment (rushilsrivastava) [inline src/webpack/CompilerEventsFacade.ts:27]: Let's release this as a minor patch, and work on refactoring this to use the new `processAssets` and release that as a major patch.

**Review by rushilsrivastava — APPROVED** [review]
This is good for a minor release, let's refactor the logic in a major release soon.

**Pull request merged**